xbmc/CueDocument.cpp

```diff
@@ -169,66 +172,71 @@ bool CCueDocument::ParseTag(const std::string &strContent)
 //////////////////////////////////////////////////////////////////////////////////
 void CCueDocument::GetSongs(VECSONGS &songs)
 {
-  const std::shared_ptr<CAdvancedSettings> advancedSettings = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings();
-
-  for (const auto& track : m_tracks)
-  {
-    CSong aSong;
-    //Pass artist to MusicInfoTag object by setting artist description string only.
-    //Artist credits not used during loading from cue sheet.
-    if (track.strArtist.empty() && !m_strArtist.empty())
-      aSong.strArtistDesc = m_strArtist;
-    else
-      aSong.strArtistDesc = track.strArtist;
-    //Pass album artist to MusicInfoTag object by setting album artist vector.
-    aSong.SetAlbumArtist(StringUtils::Split(m_strArtist, advancedSettings->m_musicItemSeparator));
-    aSong.strAlbum = m_strAlbum;
-    aSong.genre = StringUtils::Split(m_strGenre, advancedSettings->m_musicItemSeparator);
-    aSong.strReleaseDate = StringUtils::Format("{:04}", m_iYear);
-    aSong.iTrack = track.iTrackNumber;
-    if (m_iDiscNumber > 0)
-      aSong.iTrack |= (m_iDiscNumber << 16); // see CMusicInfoTag::GetDiscNumber()
-    if (track.strTitle.length() == 0) // No track information for this track!
-      aSong.strTitle = StringUtils::Format("Track {:2d}", track.iTrackNumber);
-    else
-      aSong.strTitle = track.strTitle;
-    aSong.strFileName = track.strFile;
-    aSong.iStartOffset = track.iStartTime;
-    aSong.iEndOffset = track.iEndTime;
-    if (aSong.iEndOffset)
-      // Convert offset in frames (75 per second) to duration in whole seconds with rounding
-      aSong.iDuration = CUtil::ConvertMilliSecsToSecsIntRounded(aSong.iEndOffset - aSong.iStartOffset);
-    else
-      aSong.iDuration = 0;
-
-    if (m_albumReplayGain.Valid())
-      aSong.replayGain.Set(ReplayGain::ALBUM, m_albumReplayGain);
-
-    if (track.replayGain.Valid())
-      aSong.replayGain.Set(ReplayGain::TRACK, track.replayGain);
-
-    songs.push_back(aSong);
-  }
+  const auto separator =
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_musicItemSeparator;
+
+  std::ranges::transform(
+      m_tracks, std::back_inserter(songs),
```

Comment:
I wonder if this is really necessary, it seems to me that using transform reduces code readability

Reply:
Up to the PR submitter, I’d say. Technically the change is just fine and readability is very subjective. For example, I’m used to read code using std::transform.